python/pyspark/ml/evaluation.py

```diff
@@ -394,15 +397,30 @@ def getMetricName(self):
     @keyword_only
     @since("2.3.0")
     def setParams(self, predictionCol="prediction", featuresCol="features",
-                  metricName="silhouette"):
+                  metricName="silhouette", distanceMeasure="squaredEuclidean"):
```

Comment:
Just so I'm clear, this doesn't cause any API incompatibility in Python in the way it would for Scala right?

Reply:
I am not expert enough in Python to answer this. A normal python script is interpreted at runtime, so source code compatibility should be enough, but a `.pyc` file might have the same issue in theory.
I'd love to hear the answer from someone with more experience on Python.

Reply:
Yeah these will not cause any incompatibility issues, the interpreter would automatically just pass the additional keyword arg.  These methods are actually all wrapped with `keyword_only` which make it look externally like `setParams(**kwargs)`, anyway.